# Ethan's Coded Analysis

## Libraries

In [1]:
import pandas as pd

## Load Data

In [2]:
df = pd.read_csv('../data/adult.csv')

## Looking at the Data

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32561 entries, 0 to 32560
Data columns (total 15 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   age             32561 non-null  int64 
 1   workclass       32561 non-null  object
 2   fnlwgt          32561 non-null  int64 
 3   education       32561 non-null  object
 4   education.num   32561 non-null  int64 
 5   marital.status  32561 non-null  object
 6   occupation      32561 non-null  object
 7   relationship    32561 non-null  object
 8   race            32561 non-null  object
 9   sex             32561 non-null  object
 10  capital.gain    32561 non-null  int64 
 11  capital.loss    32561 non-null  int64 
 12  hours.per.week  32561 non-null  int64 
 13  native.country  32561 non-null  object
 14  income          32561 non-null  object
dtypes: int64(6), object(9)
memory usage: 3.7+ MB


**Notes**: 
- 32,561 total columns.
- No NaN values present.
- Potential Bias: age, workclass, education, race, sex, native country, and income

In [4]:
df.head(3)

age workclass  fnlwgt     education  education.num marital.status  \
0   90         ?   77053       HS-grad              9        Widowed   
1   82   Private  132870       HS-grad              9        Widowed   
2   66         ?  186061  Some-college             10        Widowed   

        occupation   relationship   race     sex  capital.gain  capital.loss  \
0                ?  Not-in-family  White  Female             0          4356   
1  Exec-managerial  Not-in-family  White  Female             0          4356   
2                ?      Unmarried  Black  Female             0          4356   

   hours.per.week native.country income  
0              40  United-States  <=50K  
1              18  United-States  <=50K  
2              40  United-States  <=50K

**Notes**:
- ? used for unknown values

In [5]:

df.tail(3)

age workclass  fnlwgt education  education.num      marital.status  \
32558   40   Private  154374   HS-grad              9  Married-civ-spouse   
32559   58   Private  151910   HS-grad              9             Widowed   
32560   22   Private  201490   HS-grad              9       Never-married   

              occupation relationship   race     sex  capital.gain  \
32558  Machine-op-inspct      Husband  White    Male             0   
32559       Adm-clerical    Unmarried  White  Female             0   
32560       Adm-clerical    Own-child  White    Male             0   

       capital.loss  hours.per.week native.country income  
32558             0              40  United-States   >50K  
32559             0              40  United-States  <=50K  
32560             0              20  United-States  <=50K

In [6]:

df.sample(3)

age     workclass  fnlwgt     education  education.num  \
12935   49  Self-emp-inc  213354       Masters             14   
24253   47       Private  112791       HS-grad              9   
13368   20             ?  346341  Some-college             10   

           marital.status       occupation   relationship   race     sex  \
12935           Separated  Exec-managerial  Not-in-family  White    Male   
24253  Married-civ-spouse  Protective-serv        Husband  White    Male   
13368       Never-married                ?  Not-in-family  White  Female   

       capital.gain  capital.loss  hours.per.week native.country income  
12935             0             0              70  United-States   >50K  
24253             0             0              40  United-States  <=50K  
13368             0             0              35  United-States  <=50K

In [7]:
df.describe()

age        fnlwgt  education.num  capital.gain  capital.loss  \
count  32561.000000  3.256100e+04   32561.000000  32561.000000  32561.000000   
mean      38.581647  1.897784e+05      10.080679   1077.648844     87.303830   
std       13.640433  1.055500e+05       2.572720   7385.292085    402.960219   
min       17.000000  1.228500e+04       1.000000      0.000000      0.000000   
25%       28.000000  1.178270e+05       9.000000      0.000000      0.000000   
50%       37.000000  1.783560e+05      10.000000      0.000000      0.000000   
75%       48.000000  2.370510e+05      12.000000      0.000000      0.000000   
max       90.000000  1.484705e+06      16.000000  99999.000000   4356.000000   

       hours.per.week  
count    32561.000000  
mean        40.437456  
std         12.347429  
min          1.000000  
25%         40.000000  
50%         40.000000  
75%         45.000000  
max         99.000000

**Notes**:
- Age spread is 17 to 90.
- 3/4 of capital gain and loss are 0.
- Average hours per week is 40.

In [8]:
print(f"Race Value Counts:\n{df['race'].value_counts()}")

Race Value Counts:
race
White                 27816
Black                  3124
Asian-Pac-Islander     1039
Amer-Indian-Eskimo      311
Other                   271
Name: count, dtype: int64


In [9]:
print(df[df['race'] == 'White']['race'].count() / df[df['race'] != 'White']['race'].count())

5.862170706006323


**Notes**: Race of White to others is 5.86

In [10]:
print(f"Sex Value Counts:\n{df['sex'].value_counts()}")

Sex Value Counts:
sex
Male      21790
Female    10771
Name: count, dtype: int64


In [11]:
print(df[df['sex'] == 'Male']['sex'].count() / df[df['sex'] == 'Female']['sex'].count())

2.0230247887846997


**Notes**: Male to female ratio is 2.02

In [12]:
print(f"Education Value Counts:\n{df['education'].value_counts()}")

Education Value Counts:
education
HS-grad         10501
Some-college     7291
Bachelors        5355
Masters          1723
Assoc-voc        1382
11th             1175
Assoc-acdm       1067
10th              933
7th-8th           646
Prof-school       576
9th               514
12th              433
Doctorate         413
5th-6th           333
1st-4th           168
Preschool          51
Name: count, dtype: int64


**Notes**: Majority of the data at least finished high school.

In [13]:
print(f"Income Value Counts:\n{df['income'].value_counts()}")

Income Value Counts:
income
<=50K    24720
>50K      7841
Name: count, dtype: int64


**Notes**: Majority of the data population makes less than 50k

In [14]:
print(df[df['income'] == '>50K']['race'].value_counts())
df_50k_plus =  df[df['income'] == '>50K']
white_rich_count = df_50k_plus[df_50k_plus['race'] == 'White']['race'].count()
not_white_rich_count = df_50k_plus[df_50k_plus['race'] != 'White']['race'].count()
print(f"Ratio: {white_rich_count / not_white_rich_count}")

race
White                 7117
Black                  387
Asian-Pac-Islander     276
Amer-Indian-Eskimo      36
Other                   25
Name: count, dtype: int64
Ratio: 9.83011049723757


**Notes**:
- Those that do make more than 50k have a heavy class imbalance with the White race.
- Ratio of 9.83 of white 50k> income earners to none white 50k> earners.

In [15]:
df[df['education'] == 'Preschool'].sample(20)

age  workclass  fnlwgt  education  education.num  \
27295   52    Private  370552  Preschool              1   
11268   54          ?  148657  Preschool              1   
6718    71    Private  235079  Preschool              1   
20267   24    Private  243368  Preschool              1   
28939   36    Private  252231  Preschool              1   
1153    40    Private  566537  Preschool              1   
17344   21    Private  243368  Preschool              1   
26087   48    Private  209182  Preschool              1   
13209   59    Private  157305  Preschool              1   
27931   34  Local-gov  144182  Preschool              1   
32381   46    Private  139514  Preschool              1   
18599   75    Private   71898  Preschool              1   
26799   19    Private  277695  Preschool              1   
1106    26    Private  322614  Preschool              1   
31891   31  State-gov   77634  Preschool              1   
26554   42    Private  144995  Preschool              1   
1678    27    Private  211032  Preschool              1   
25737   32    Private  223212  Preschool              1   
15748   68    Private  168794  Preschool              1   
10198   54    Private  286989  Preschool              1   

              marital.status         occupation    relationship  \
27295     Married-civ-spouse  Machine-op-inspct         Husband   
11268     Married-civ-spouse                  ?            Wife   
6718                 Widowed       Craft-repair       Unmarried   
20267          Never-married    Farming-fishing   Not-in-family   
28939          Never-married  Machine-op-inspct   Not-in-family   
1153      Married-civ-spouse      Other-service         Husband   
17344          Never-married    Farming-fishing   Not-in-family   
26087              Separated      Other-service       Unmarried   
13209          Never-married  Machine-op-inspct   Not-in-family   
27931          Never-married       Adm-clerical       Own-child   
32381     Married-civ-spouse  Machine-op-inspct  Other-relative   
18599          Never-married    Priv-house-serv   Not-in-family   
26799          Never-married    Farming-fishing   Not-in-family   
1106   Married-spouse-absent  Machine-op-inspct   Not-in-family   
31891          Never-married      Other-service   Not-in-family   
26554          Never-married  Handlers-cleaners   Not-in-family   
1678      Married-civ-spouse    Farming-fishing  Other-relative   
25737     Married-civ-spouse  Machine-op-inspct         Husband   
15748          Never-married  Machine-op-inspct   Not-in-family   
10198     Married-civ-spouse    Farming-fishing         Husband   

                     race     sex  capital.gain  capital.loss  hours.per.week  \
27295               White    Male             0             0              40   
11268               White  Female             0             0              40   
6718                Black    Male             0             0              10   
20267               White    Male             0             0              36   
28939               Black    Male             0             0              40   
1153                White    Male             0          1672              40   
17344               White    Male             0             0              50   
26087               White  Female             0             0              40   
13209               White    Male             0             0              40   
27931               Black  Female             0             0              25   
32381               Black    Male             0             0              75   
18599  Asian-Pac-Islander  Female             0             0              48   
26799               White    Male             0             0              36   
1106                White    Male             0          1719              40   
31891               White    Male             0             0              24   
26554               White    Male             0             0  

**Notes**: Based on this sample, occupation could correlate to education level.